src/Components/Common/components/Menu.tsx

```diff
@@ -0,0 +1,59 @@
+import { Menu } from "@headlessui/react";
+import { DropdownTransition } from "./HelperComponents";
+
+export default function DropdownMenu(props: {
+  title: string;
+  icon: JSX.Element;
+  children: JSX.Element | JSX.Element[];
+}) {
+  return (
+    <div className="text-right">
+      <Menu as="div" className="relative inline-block text-left w-full">
+        <Menu.Button className="flex w-full lg:justify-between justify-center items-center button-primary-default Button outline-offset-1 button-size-default button-shape-square">
+          <div className="flex gap-2 items-center">
+            {props.icon}
+            {props.title}
+          </div>
+
+          <i className="ml-2 -mr-1 text-violet-200 hover:text-violet-100 uil uil-angle-down text-2xl font-bold"></i>
+        </Menu.Button>
+        <DropdownTransition>
+          <Menu.Items className="absolute right-0  mt-2 py-1 min-w-full sm:min-w-[250px] origin-top-right rounded-[4px] bg-white shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none">
+            <>{props.children}</>
+          </Menu.Items>
+        </DropdownTransition>
+      </Menu>
+    </div>
+  );
+}
+
+export function DropdownItem(props: {
+  children: JSX.Element | JSX.Element[];
+  activeColor?: string;
+}) {
+  return (
+    <Menu.Item>
+      {({ active }) => (
+        <div
+          style={{
+            background: `${
+              active
+                ? props.activeColor
+                  ? props.activeColor + "1A"
+                  : "#0D9F6E1A"
+                : "#FFFFFF"
+            }`,
+            color: active
+              ? props.activeColor
+                ? props.activeColor
+                : "#0D9F6E"
+              : "#111827",
+          }}
+          className="rounded-[4px] m-2 px-4 py-2 text-sm cursor-pointer"
+        >
+          <>{props.children}</>
+        </div>
+      )}
+    </Menu.Item>
+  );
+}
```

Comment:
You can lock the design flexibility by using variants (similar to how button v2 is implemented) so that we achieve consistent design across the app but provide design flexibility based on the context.

1. Accept `variant: ButtonVariant` prop instead of `activeColor` and default it to `"primary".
![image](https://user-images.githubusercontent.com/25143503/200118160-c24f42be-0a1a-456e-9519-651629421dcd.png)

2. Apply colours styles based on the variant
You might be able to directly use the `button-[primary/secondary/...]-default` classes along and append `hover:bg-opacity-10` to achieve the style.
![image](https://user-images.githubusercontent.com/25143503/200118271-7b18cce2-91a4-491a-b0f6-51a094d35336.png)
![image](https://user-images.githubusercontent.com/25143503/200118303-b57d708c-6268-4e1f-b542-39641d0e51ce.png)